src/views/Merge.vue

```diff
@@ -1,94 +1,156 @@
-		const appendPDF = await PDFDocument.load(this.files[i].data);
-		const appendPages = await mergePDF.copyPages(appendPDF, appendPDF.getPageIndices());
-		for(let i = 0; i < appendPages.length; i++){
-			await mergePDF.addPage(appendPages[i]);
-		}
-	}
+            // https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop
 
-	const mergePDFFile = await mergePDF.save();
-	download(mergePDFFile, "merged.pdf", "application/pdf");
-	this.isLoading = false;
-},
-mosaicPDF: async function(){
-	let xC = 2;
-	let yC = 5;
+            for await (const file of this.files)
+            {
+                const appendPDF = await PDFDocument.load(file.data);
+                const appendPages = await mergePDF.copyPages(appendPDF, appendPDF.getPageIndices());
 
-	const mosaicPDF = await PDFDocument.create();
-	const embedPdf = await PDFDocument.load(this.files[0].data);
-	const embedPages = await mosaicPDF.embedPdf(embedPdf, embedPdf.getPageIndices());
+                appendPages.forEach(appendPage => mergePDF.addPage(appendPage));
+            }
 
-	for(let i = 0; i < embedPages.length; i += xC + yC){
-		const page = mosaicPDF.addPage();
-		for(let x = 0; x < xC; x++){
-			for(let y = 0; y < yC; y++){
-				if(x + xC * y > embedPages.length) break;
-				page.drawPage(embedPages[x + xC * y], {
-					x: x * (page.getWidth()/xC),
-					y: page.getHeight()*((yC-1)/yC) - (page.getHeight() / yC * y),
-					width: page.getWidth()/xC,
-					height: page.getHeight()/yC,
-				});
-			}
-		}
-	}
-	const mergePDFFile = await mosaicPDF.save();
-	download(mergePDFFile, "merged.pdf", "application/pdf");
-}
-}
+            const mergePDFFile = await mergePDF.save();
+            download(mergePDFFile, "merged.pdf", "application/pdf");
+            this.isLoading = false;
+        },
+        mosaicPDF: async function(){
+            let xC = 2;
+            let yC = 5;
+
+            const mosaicPDF = await PDFDocument.create();
+            const embedPdf = await PDFDocument.load(this.files[0].data);
+            const embedPages = await mosaicPDF.embedPdf(embedPdf, embedPdf.getPageIndices());
+
+            for(let i = 0; i < embedPages.length; i += xC + yC){
+                const page = mosaicPDF.addPage();
+                for(let x = 0; x < xC; x++){
+                    for(let y = 0; y < yC; y++){
+                        if(x + xC * y > embedPages.length) break;
+                        page.drawPage(embedPages[x + xC * y], {
+                            x: x * (page.getWidth()/xC),
+                            y: page.getHeight()*((yC-1)/yC) - (page.getHeight() / yC * y),
+                            width: page.getWidth()/xC,
+                            height: page.getHeight()/yC,
+                        });
+                    }
+                }
+            }
+            const mergePDFFile = await mosaicPDF.save();
+            download(mergePDFFile, "merged.pdf", "application/pdf");
+        },
```

Comment:
Something is broken here, but it's not your fault. It was broken since before you forked it.

Reply:
what is broken exactly? I don't get any error when I run it.

Reply:
The grid mode is broken at least for me, see #7 for the log message.

Reply:
It seems it is not working only with some pdfs. I got this error with only one pdf. All the others where working fine, even bigger ones.

Reply:
weird, didn't work for any pdfs I tried 🤔